[openbsd] build problem - error: expected specifier-qualifier-list before 'time_t'

I tried to build 3.6.2 on OpenBSD 5.7 amd64 snapshost (Fri Apr 24 12:59:17 MDT 2015) and it fails because of time_t.

```
cc -std=gnu99 -DHAVE_CONFIG_H -I. -I./lib -I./modules -I./lib -I./modules -pthread -I/usr/local/include/glib-2.0 -I/usr/local/lib/glib-2.0/include -I/usr/local/include -I/usr/local/include/eventlog -I/usr/local/include -I/usr/include/openssl -I/usr/include/openssl -I/usr/local/include -D_GNU_SOURCE -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -O2 -pipe -O0 -g -I/usr/local/include -Wall -pthread -MT lib/compat/lib_libsyslog_ng_la-getutent.lo -MD -MP -MF lib/compat/.deps/lib_libsyslog_ng_la-getutent.Tpo -c lib/compat/getutent.c -fPIC -DPIC -o lib/compat/.libs/lib_libsyslog_ng_la-getutent.o
In file included from ./lib/compat/getutent.h:34,
                 from lib/compat/getutent.c:25:
/usr/include/utmp.h:57: error: expected specifier-qualifier-list before 'time_t'
/usr/include/utmp.h:66: error: expected specifier-qualifier-list before 'time_t'
Error while executing cc -std=gnu99 -DHAVE_CONFIG_H -I. -I./lib -I./modules -I./lib -I./modules -pthread -I/usr/local/include/glib-2.0 -I/usr/local/lib/glib-2.0/include -I/usr/local/include -I/usr/local/include/eventlog -I/usr/local/include -I/usr/include/openssl -I/usr/include/openssl -I/usr/local/include -D_GNU_SOURCE -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -O2 -pipe -O0 -g -I/usr/local/include -Wall -pthread -MT lib/compat/lib_libsyslog_ng_la-getutent.lo -MD -MP -MF lib/compat/.deps/lib_libsyslog_ng_la-getutent.Tpo -c lib/compat/getutent.c -fPIC -DPIC -o lib/compat/.libs/lib_libsyslog_ng_la-getutent.o
Makefile:7472: recipe for target 'lib/compat/lib_libsyslog_ng_la-getutent.lo' failed
gmake[2]: *** [lib/compat/lib_libsyslog_ng_la-getutent.lo] Error 2
Makefile:9846: recipe for target 'all-recursive' failed
gmake[1]: *** [all-recursive] Error 1
Makefile:4042: recipe for target 'all' failed
gmake: *** [all] Error 2
*** Error 2 in . (/home/jirib/cvs/openbsd-ports/infrastructure/mk/bsd.port.mk:2755 '/home/jirib/openbsd/pobj/syslog-ng-3.6.2/.build_done')
*** Error 1 in /home/jirib/git/openbsd-mystuff/sysutils/syslog-ng (/home/jirib/cvs/openbsd-ports/infrastructure/mk/bsd.port.mk:2477 'all')
```

http://cvsweb.openbsd.org/cgi-bin/cvsweb/~checkout~/src/include/utmp.h?rev=1.6

```
     56 struct lastlog {
     57         time_t  ll_time;
     58         char    ll_line[UT_LINESIZE];
     59         char    ll_host[UT_HOSTSIZE];
     60 };
     61
     62 struct utmp {
     63         char    ut_line[UT_LINESIZE];
     64         char    ut_name[UT_NAMESIZE];
     65         char    ut_host[UT_HOSTSIZE];
     66         time_t  ut_time;
     67 };
     68
     69 #endif /* !_UTMP_H_ */
```

--- a/lib/compat/getutent.h
+++ b/lib/compat/getutent.h
@@ -27,6 +27,7 @@
 
 #include "compat/compat.h"
 #include <inttypes.h>
+#include <time.h>
 
 #ifdef HAVE_UTMPX_H
 #include <utmpx.h>

--- a/lib/compat/getutent.h
+++ b/lib/compat/getutent.h
@@ -25,8 +25,8 @@
 #ifndef COMPAT_GETUTENT_H_INCLUDED
 #define COMPAT_GETUTENT_H_INCLUDED
 
-#include <inttypes.h>
 #include "compat/compat.h"
+#include <inttypes.h>
 
 #ifdef HAVE_UTMPX_H
 #include <utmpx.h>

